h2o-py/tests/testdir_misc/pyunit_uniop_colname_domain.py

```diff
@@ -0,0 +1,23 @@
+from __future__ import print_function
+import sys
+sys.path.insert(1,"../../")
+import h2o
+from tests import pyunit_utils
+
+#Test to see if you apply H2O uniop methods to an H2OFrame it will not overwrite the original column headers & domains
+# & also update colnames of new frame with `op(colName)` where `op` is a uniop operator.
+
+def pyunit_colname_uniop():
+    dataframe = {'A': [1,0,3,4], 'B': [5,6,-6, -1], 'C':[-4, -6, -7, 8]}
+    frame = h2o.H2OFrame(dataframe)
+    frame_asin = frame.asin()
+
+    #Check colnames of original frame remain untouched.
+    assert frame.names == [u'A', u'C', u'B'], "Expected original colnames to remain the same after uniop operation"
+    #Check new colnames for modified frame are of the convention `op(colname)`
+    assert ["asin(%s)" % (name) for name in frame.names] == frame_asin.names,"Expected equal col names after uniop operation"
+
```

Comment:
Note that the frame `frame_asin` remained unexecuted. You need to force its execution (for example by printing it or saying `frame_asin.refresh()`), and then test the same invariants as above once again.